appknox/client.py

```diff
@@ -430,9 +430,21 @@ def upload_file(self, file_data: str):
                 url=response['url']
             )
         )
+        submission_id = response2['submission_id']
+        return self.get_file_from_submission_id(submission_id)
+
+    def get_file_from_submission_id(self, submission_id: int):
+        """
+        Using the submission id, keep checking its status.
+        Returns file instance when it's available
+
+        :param submission_id: The ID of the submission object
+         created for the scan
+
+         :return: The File ID
+        """
         file = None
         timeout = time.time() + 10
-        submission_id = response2['submission_id']
         while (file is None):
             submission = self.drf_api.submissions(submission_id).get()
             if submission.get('detail') == 'Not found.':
```

Comment:
This util method should not be throwing `UploadError` instead throw unique errors for each scenario. Then we can catch those errors in `upload_file` & `rescan` to return proper error messages.

Example:

```python
while (file is None):
    submission = self.drf_api.submissions(submission_id).get()
    if submission.get('detail') == 'Not found.':
        raise SubmissionNotFound()
    submission_obj = mapper_drf_api(Submission, submission)
    file = submission_obj.file
    if submission_obj.reason:
        raise SubmissionError(submission_obj.reason)
    if time.time() > timeout:
        raise SubmissionFileTimeoutError()
```

In `upload_file ` we can use as:
```python
try:
	self.get_file_from_submission_id(submission_id)
except SubmissionNotFound:
	raise UploadError('Something went wrong, try uploading the file again.')
...
```

And in `rescan `:
```python
try:
	self.get_file_from_submission_id(submission_id)
except SubmissionNotFound:
	raise RescanError('Something went wrong, retry rescan')
...
```